src/components/_button.scss

```diff
@@ -59,210 +59,145 @@ $btn-padding-x: (
     width: map-get(variables.$form-controls-height, sm);
   }
 
-  &.primary {
-    background-color: colors.$primary;
-    color: #fff;
-    border-bottom-color: colors.$primary-dark;
-
-    &:hover,
-    &:focus {
-      background-color: colors.$primary-hover;
-    }
-
-    &:focus {
-      box-shadow: 0px 0px 0px 3px rgba(48, 79, 253, 0.3);
-    }
-
-    &:active {
-      background-color: colors.$primary-dark;
-    }
-  }
-
-  &.warning {
-    background-color: colors.$warning;
-    color: #fff;
-    border-bottom-color: colors.$warning-dark;
-
-    &:hover,
-    &:focus {
-      background-color: colors.$warning-hover;
-    }
-
-    &:focus {
-      box-shadow: 0px 0px 0px 3px rgba(229, 110, 25, 0.3);
-    }
-
-    &:active {
-      background-color: colors.$warning-dark;
-    }
-  }
-
-  &.danger {
-    background-color: colors.$danger;
-    color: #fff;
-    border-bottom-color: colors.$danger-dark;
-
-    &:hover,
-    &:focus {
-      background-color: colors.$danger-hover;
-    }
-
-    &:focus {
-      box-shadow: 0px 0px 0px 3px rgba(253, 121, 114, 0.3);
-    }
-
-    &:active {
-      background-color: colors.$danger-dark;
-    }
-  }
-
-  &.success {
-    background-color: colors.$success;
-    color: #fff;
-    border-bottom-color: colors.$success-dark;
-
-    &:hover,
-    &:focus {
-      background-color: colors.$success-hover;
-    }
-
-    &:focus {
-      box-shadow: 0px 0px 0px 3px rgba(73, 201, 109, 0.3);
-    }
-
-    &:active {
-      background-color: colors.$success-dark;
-    }
-  }
-
-  &.light {
-    background-color: #fff;
-    border-bottom-color: transparent;
-
-    &:hover {
-      background-color: colors.$bg-dark;
-    }
-
-    &.primary {
-      background-color: colors.$primary-light;
-      color: colors.$primary;
+  @mixin button(
+    $color,
+    $bg: colors.$primary,
+    $hover: colors.$primary-hover,
+    $active: colors.$primary-dark,
+    $focus: colors.$primary-shadow
```

Comment:
Why do you need default values here? Is there a possibility that this mixin will be called frequently without arguments passed?